### PR TITLE
(DOCSP-29536): C++: update test suite to v0.1.1-alpha

### PR DESCRIPTION
**Pull request by dacharyc**
## Pull Request Info

Updating the C++ test suite to use versioned releases!

A few minor code changes to fix build errors, but the only visible changes are listed in the Staged Changes section.

### Jira

- https://jira.mongodb.org/browse/DOCSP-29536

### Staged Changes

- [Handle Sync Errors](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-29536/sdk/cpp/sync/handle-sync-errors/): Remove `error_code()` as it is no longer available 
- [Threading](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-29536/sdk/cpp/crud/threading/#schedulers--run-loops-): Overriding `invoke` in a custom scheduler now requires `realm::Function` instead of `std::function`. I also added some minimal code to get past build errors, but that code is not reflected in the published code snippet.

### Reminder Checklist

If your PR modifies the docs, you might need to also update some corresponding
pages. Check if completed or N/A.

- [x] Create Jira ticket for corresponding docs-app-services update(s), if any
- [x] Checked/updated Admin API
- [x] Checked/updated CLI reference

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: examples/cpp/quick-start.cpp [hunk 1]
```diff
@@ -101,6 +101,7 @@ TEST_CASE("local quick start", "[realm][write]") {
         realm.remove(todo);
     });
     // :snippet-end:
+    token.unregister();
```
Comment (cbush) [inline examples/cpp/quick-start.cpp:104]: Why is this needed?

File: examples/cpp/threading.cpp [hunk 2]
```diff
@@ -63,20 +63,29 @@ TEST_CASE("scheduler", "[write]") {
...
         }
         
-        void invoke(std::function<void()> &&task) override {
+        void invoke(realm::Function<void()> &&task) override {
             // ... Add the task to the (lock-free) processor queue ...
+            // :remove-start:
+            // Had to implement some code in here in order to override invoke
+            // This is arbitrary code that isn't shown in the example just to
+            // get the tests to compile and run.
+            task.release();
```
Comment (cbush) [inline examples/cpp/threading.cpp:72]: Why is this needed?

File: examples/cpp/quick-start.cpp (hunk 1 above)
Comment (dacharyc) [reply examples/cpp/quick-start.cpp:104]: After upgrading the test suite to the new version, subsequent tests were failing with an error that there was already a realm open with a different schema. Unregistering the observation token appears to let the realm properly go out of scope and resolves test failures in those subsequent tests.

File: examples/cpp/threading.cpp (hunk 2 above)
Comment (dacharyc) [reply examples/cpp/threading.cpp:72]: With no executable code in this overridden function, `make` was failing to build the test suite with an error that this was not a valid override. Adding executable code allowed the test to build, and this seemed benign.

File: examples/cpp/quick-start.cpp (hunk 1 above)
Comment (cbush) [reply examples/cpp/quick-start.cpp:104]: The fact that the token doesn't unregister itself when going out of scope appears to be a violation of [RAII](https://en.cppreference.com/w/cpp/language/raii). This would be an SDK issue. Worth raising with the team?

File: examples/cpp/threading.cpp (hunk 2 above)
Comment (cbush) [reply examples/cpp/threading.cpp:72]: Can you post the full error? This doesn't seem right, and I don't want to get in the habit of adding meaningless code.

Comment (dacharyc) [reply examples/cpp/threading.cpp]: No, you're right! I guess the invalid override message was cleared by changing `std::function` to `realm::Function` and I did not realize that was the change that resolved the error. I was able to remove this and the test still passes. Thank you for pushing!

**Review by cbush — APPROVED** [review]
